config/oidc/config.proto

```diff
@@ -75,14 +75,30 @@ message OIDCConfig {
     // Required.
     string callback_uri = 3 [(validate.rules).string.min_len = 1];
 
-    // The JSON JWKS response from the OIDC provider’s `jwks_uri` URI which can be found in
-    // the OIDC provider's
-    // [configuration response](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse).
-    // Note that this JSON value must be escaped when embedded in a json configmap
-    // (see [example](https://github.com/istio-ecosystem/authservice/blob/master/bookinfo-example/config/authservice-configmap-template.yaml)).
-    // Used during token verification.
-    // Required.
-    string jwks = 4 [(validate.rules).string.min_len = 1];
+    // This message defines a setting to allow asynchronous retrieval and update of the JWK for
+    // JWT validation at regular intervals.
+    message JwksFetcherConfig {
+        // Request URI that has the JWKs.
+        // Required.
+        string jwks_uri = 1;
+        
+        // Request interval to check whether new JWKs are available.
+        // Required.
+        int32 request_duration_sec = 2;
+    }
+
+    oneof jwks_config {
+        // The JSON JWKS response from the OIDC provider’s `jwks_uri` URI which can be found in
+        // the OIDC provider's
+        // [configuration response](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse).
+        // Note that this JSON value must be escaped when embedded in a json configmap
+        // (see [example](https://github.com/istio-ecosystem/authservice/blob/master/bookinfo-example/config/authservice-configmap-template.yaml)).
+        // Used during token verification.
+        string jwks = 4;
+
+        // Configuration to allow JWKs to be retrieved and updated asynchronously at regular intervals.
+        JwksFetcherConfig jwks_fetcher = 17;
```

Comment:
should we add a NEXT tag in the beginning of the proto to reminds future editing what's next available number? smt like https://github.com/envoyproxy/envoy/blob/main/api/envoy/data/cluster/v2alpha/outlier_detection_event.proto#L77

Reply:
These free field is generated automatically on envoy with complex protoxform scripting. But it wiil be difficult to automate them in this tiny repository. But it will be good to have manually. WDYT?